### ISSUE TITLE
Adding new options makes editing old entries not work.

### ISSUE THREAD

**Issue by BrandonSurowiec** [fourstacks/nova-checkboxes#14]
When the options array gets updated, all old options saved in the database are not handle properly and the options cannot be checked in Chrome.

Given you had the options:

```
->options([
     'a' => 'Option A',
    'b' => 'Option B',
])

// Create an entry in the database
// {a: true, b:false}

```

But then you modify the options field:

```
->options([
                'c' => 'Option C',
                'd' => 'Option D',
])
```

When going to edit the resources, the options don't select properly. It stays blank and shows a blue outline. This is because the value is passed through as `{a: true, b:false}` instead of `{c: false, d:false}`.

To fix this, the resolveAttribute function needs to apply ->withUnchecked for the associate array:

```
            if ($this->arrayIsAssoc($value)) {
                return json_encode($this->withUnchecked($value));
            }
```

and `function withUnchecked()` needs to cross reference the values from the database with the values in the options:

```
            ->mapWithKeys(function ($option) use ($data) {
                $isChecked = $data[$option['value']] ?? false;

                return [ $option['value'] => $isChecked ];
            })
```

I would change this and submit a PR, but because this has no tests, I wouldn't feel comfortable submitting the PR until I had the time to write tests to verify the different inputs and output values.

Also, I recommend switching all `private` method calls to be `protected`, so if someone were to extend the class, they could override your methods. I extended the class to fix it for my use case temporarily but had to come up with some other function names since your private methods were unable to be overridden.

I'll see if I can carve out some time this later week to help fix if you're unable to get to it sooner.

**Comment by fourstacks** [comment]
@BrandonSurowiec thanks very much for the bug report and suggested fix.  

Yeah sorry for lack of tests.  I needed this package in a hurry when I developed it and didn't have the time to sink into working out the best way to test nova fields.  Should really come back round to that at some point...

You were on the right track with your fix but it did affect some fo the other save types so I ended up achieving a fix that works across all save types by using the existing `onlyChecked` function to convert the associative array to a simple array containing only the checked items.  This then feeds through the `withUnchecked` function (which remains unchanged), fixing the issue.  No 'new' code in this fix which is nice.

Thanks for the suggestion of the moving private -> protected.  Have implemented that too.

If you could do me a favour and pull down `dev-master` and let me know that it fixes the issue you were having then I'll tag a new release.  Thanks

**Comment by BrandonSurowiec** [comment]
@fourstacks Thank you for the quick patch John! I pulled in `dev-master` and it did indeed fix my issue. I completely understand about the tests. I'll do what I can later this week/weekend to help contribute some. Thank you again!

**Comment by fourstacks** [comment]
👍 No worries!

**Comment by fourstacks** [comment]
Fixed in v0.1.3
